File: 0209-minimum-size-subarray-sum/solution.cpp
class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
        int s = 0;
        int tempSum = 0;
        int minLength = INT_MAX;
        for(int i=0; i<nums.size(); i++){
            tempSum +=nums[i];

            if(tempSum == target){
                minLength = min(minLength, i-s+1);
            }

            while(tempSum >= target){
                minLength = min(minLength, i-s+1);
                tempSum -= nums[s];
                s++;
            }
        }
        if(minLength == INT_MAX) return 0;
        return minLength;
    }
};

File: 0209-minimum-size-subarray-sum/solution.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int sum = 0;
        int left = 0;
        int miniSize = Integer.MAX_VALUE;
        for(int i=0;i<nums.length; i++){
            sum+=nums[i];

            while(left <= i && sum >= target){
                miniSize = Math.min(miniSize, i-left+1);
                sum-=nums[left++];
            }
        }
        if(miniSize == Integer.MAX_VALUE) return 0;
        return miniSize;
    }
}
